### PR TITLE
Adding Google DNS module

### PR DESCRIPTION
**Pull request by erjohnso**
New module to add support for Google DNS [https://cloud.google.com/dns/]. This is modeled very closely to the route53 module.

As with the other Google modules, this is implemented using libcloud and a safety import-check is included.

### CODE REVIEW

File: cloud/gc_dns.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/python
```
Comment (fcuny) [inline cloud/gc_dns.py:1]: could you use `/usr/bin/env python` instead ? That way it also works nicely with virtualenv

File: cloud/gc_dns.py [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+            value = dict(required=False),
+            pem_file = dict(),
+            service_account_email = dict(),
+            project_id = dict(),
+        )
+    )
+    conn = gce_connect(module, Provider.GOOGLE)
+    # TODO: figure out why this is needed to prevent SSL-required error
+    conn.connection.secure = True
+    conn.connection.port = 443
```
Comment (fcuny) [inline cloud/gc_dns.py:201]: This was due to a bug in libcloud. I've submitted a fix over there.

**Comment by fcuny** [comment]
LGTM!

**Comment by erjohnso** [comment]
Per agreement with @mpdehaan, starting this out in extras.

**Pull request closed without merging**